### DISCUSSION TITLE
[ 已完成 ][ 设置 - 功能管理 ] 能否只作为本地播放器使用

### DISCUSSION THREAD

**Discussion by wayner6** [GeekLee2012/Less-Player-Desktop#17] [Ideas]
即去除其他的音乐平台，去除仅保留本地歌曲功能
![image](https://github.com/GeekLee2012/Less-Player-Desktop/assets/88063517/479689b4-a2c9-4802-b1ea-3fc67f6b24c8)
可以加一个开关，来选择性关闭这些功能嘛？

**Comment by GeekLee2012** [comment]
朋友你好！单纯仅实现这点需求，应该很简单。

但转念细想，从全局角度看，选择性开启/关闭功能，这个需求其实并不是表面这么简单。
因为，关闭一个功能点，会涉及其他相关多个功能点。
当然，也可以简单粗暴一些，仅作隐藏，不去检查和处理其他关联的功能点。

应该有不少朋友，会有这类需求。
我再想想具体的实现方式，尽可能灵活一些吧。

最后再多说两句，
功能虽然选择性关闭了，但播放器大小、性能方面并不会有什么变化。
只是“眼不见，心不烦”罢了，哈哈哈。

**Comment by wayner6** [comment]
如果做个开关不容易的话，其实可以直接发行一个本地版的，我很喜欢您软件的ui和逻辑，但是网易云，qq那些是真的用不到，看着也烦哈哈，本地播放器唯一的联网需求就是根据歌曲搜索封面和歌词😊😊

**Comment by GeekLee2012** [comment]
单独发行本地版播放器，一个人可忙不过来。
而且从现有功能来看，作为独立版本，略显单薄；下个小版本，我再继续完善一下本地歌曲：增强搜索、数据管理这两个方面。
扯远啦，单独版本，目前来说不太可能的。

还是按照当前的UI方式，设置页里新增一些开关选项吧。
每多加一个开关，可能会相应多出一个逻辑分支......


现在本地歌曲的一些逻辑：
- 歌词优先级：内嵌歌词 > 本地同名歌词文件（仅支持.lrc格式） >  在线歌词
- 歌曲封面逻辑：如果“设置”里启用在线封面，在线封面会替换内置封面

**Comment by wayner6** [comment]
好的，谢谢开发者的耐心回复，永远支持您的创作😁
